src/Entities/Coupons/Restriction.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * This file is part of the PHP Rebilly API package.
+ *
+ * (c) 2015 Rebilly SRL
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Rebilly\Entities\Coupons;
+
+use DomainException;
+use Rebilly\Rest\Resource;
+
+class Restriction extends Resource
+{
+    const MSG_UNSUPPORTED_TYPE = 'Unexpected type. Only %s types support';
+    const MSG_REQUIRED_TYPE = 'Type is required';
+
+    const TYPE_DISCOUNTS_PER_REDEMPTION = 'discounts-per-redemption';
+    const TYPE_MINIMUM_ORDER_AMOUNT = 'minimum-order-amount';
+    const TYPE_REDEMPTIONS_PER_CUSTOMER = 'redemptions-per-customer';
+    const TYPE_RESTRICT_TO_INVOICES = 'restrict-to-invoices';
+    const TYPE_RESTRICT_TO_PLANS = 'restrict-to-plans';
+    const TYPE_RESTRICT_TO_PRODUCTS = 'restrict-to-products';
+    const TYPE_RESTRICT_TO_SUBSCRIPTIONS = 'restrict-to-subscriptions';
+    const TYPE_TOTAL_REDEMPTIONS = 'total-redemptions';
+
+    protected static $supportedTypes = [
+        self::TYPE_DISCOUNTS_PER_REDEMPTION,
+        self::TYPE_MINIMUM_ORDER_AMOUNT,
+        self::TYPE_REDEMPTIONS_PER_CUSTOMER,
+        self::TYPE_RESTRICT_TO_INVOICES,
+        self::TYPE_RESTRICT_TO_PLANS,
+        self::TYPE_RESTRICT_TO_PRODUCTS,
```

Comment:
`self::TYPE_RESTRICT_TO_PRODUCTS`, `self::TYPE_MINIMUM_ORDER_AMOUNT`, `self::TYPE_TOTAL_REDEMPTIONS` are not supported yet.